satellite/satellitedb/repairqueue.go

```diff
@@ -72,33 +70,35 @@ func (r *repairQueue) sqliteDequeue(ctx context.Context) (seg pb.InjuredSegment,
 	return seg, err
 }
 
-func (r *repairQueue) Dequeue(ctx context.Context) (seg pb.InjuredSegment, err error) {
+func (r *repairQueue) Select(ctx context.Context) (seg *pb.InjuredSegment, err error) {
 	switch t := r.db.DB.Driver().(type) {
 	case *sqlite3.SQLiteDriver:
-		return r.sqliteDequeue(ctx)
+		return r.sqliteSelect(ctx)
 	case *pq.Driver:
-		return r.postgresDequeue(ctx)
+		return r.postgresSelect(ctx)
 	default:
 		return seg, fmt.Errorf("Unsupported database %t", t)
 	}
 }
 
-func (r *repairQueue) Peekqueue(ctx context.Context, limit int) ([]pb.InjuredSegment, error) {
+func (r *repairQueue) Delete(ctx context.Context, seg *pb.InjuredSegment) (err error) {
+	_, err = r.db.ExecContext(ctx, r.db.Rebind(`DELETE FROM injuredsegments WHERE path = ?`), seg.Path)
+	return
+}
+
+func (r *repairQueue) SelectN(ctx context.Context, limit int) (segs []pb.InjuredSegment, err error) {
 	if limit <= 0 || limit > storage.LookupLimit {
 		limit = storage.LookupLimit
 	}
-	rows, err := r.db.Limited_Injuredsegment(ctx, limit, 0)
-	if err != nil {
-		return nil, err
-	}
-
-	segments := make([]pb.InjuredSegment, 0)
-	for _, entry := range rows {
-		seg := &pb.InjuredSegment{}
-		if err = proto.Unmarshal(entry.Info, seg); err != nil {
-			return nil, err
+	//todo: strictly enforce order-by or change tests
+	rows, err := r.db.QueryContext(ctx, r.db.Rebind(`SELECT data FROM injuredsegments LIMIT ?`), limit)
+	for rows.Next() {
+		var seg pb.InjuredSegment
+		err = rows.Scan(&seg)
+		if err != nil {
+			return
 		}
-		segments = append(segments, *seg)
+		segs = append(segs, seg)
 	}
-	return segments, nil
+	return
```

Comment:
We should inspect [`rows.Err`](https://golang.org/pkg/database/sql/#Rows.Err) before returning